pkg/apis/batch/validation/validation.go

```diff
@@ -139,11 +139,16 @@ func validateGeneratedSelector(obj *batch.Job, validateBatchLabels bool) field.E
 }
 
 // ValidateJob validates a Job and returns an ErrorList with any errors.
-func ValidateJob(job *batch.Job, opts JobValidationOptions) field.ErrorList {
+func ValidateJob(job, oldJob *batch.Job, opts JobValidationOptions) field.ErrorList {
```

Comment:
I don't really understand why this change is needed for this PR?

Reply:
We need all old resource objects containing PodSpec so that we can judge whether the updating object use relaxed or strict validation mode.

Reply:
Sounds good.